src/common/config/constants.ts

```diff
@@ -38,106 +38,108 @@ export const getProductBrandingCode = () => 'redhat';
 export const POLLING_INTERVAL = 10 * 1000;
 export const EVENTS_POLLING_INTERVAL = 10 * 1000;
 
-export const HOST_ROLES: HostRole[] = [
+export const hostRoles = (t: TFunction): HostRole[] => [
   {
-    value: 'auto-assign',
-    label: 'Auto-assign',
-    description:
-      'A role will be chosen automatically based on detected hardware and network latency.',
+    value: t('ai:auto-assign'),
+    label: t('ai:Auto-assign'),
+    description: t(
+      'ai:A role will be chosen automatically based on detected hardware and network latency.',
+    ),
   },
   {
-    value: 'master',
-    label: 'Control plane node',
-    description: 'Runs the control plane components of OpenShift, including the API server.',
+    value: t('ai:master'),
+    label: t('ai:Control plane node'),
+    description: t('ai:Runs the control plane components of OpenShift, including the API server.'),
   },
   {
-    value: 'worker',
-    label: 'Worker',
-    description:
-      'Runs application workloads. Connect at least 5 hosts to enable dedicated workers.',
+    value: t('ai:worker'),
+    label: t('ai:Worker'),
+    description: t(
+      'ai:Runs application workloads. Connect at least 5 hosts to enable dedicated workers.',
+    ),
   },
 ];
 
-export const CLUSTER_STATUS_LABELS: { [key in Cluster['status']]: string } = {
-  'pending-for-input': 'Draft',
-  insufficient: 'Draft',
-  ready: 'Draft',
-  'preparing-for-installation': 'Preparing for installation',
-  installing: 'Installing',
-  'installing-pending-user-action': 'Installing (pending action)',
-  finalizing: 'Finalizing',
-  cancelled: 'Installation cancelled',
-  error: 'Error',
-  installed: 'Installed',
-  'adding-hosts': 'Adding hosts',
-};
-
-export const CLUSTER_FIELD_LABELS: { [key in string]: string } = {
-  name: 'Cluster name',
-  baseDnsDomain: 'Base domain',
-  clusterNetworkCidr: 'Cluster network CIDR',
-  clusterNetworkHostPrefix: 'Cluster network host prefix',
-  serviceNetworkCidr: 'Service network CIDR',
-  apiVip: 'API IP',
-  ingressVip: 'Ingress IP',
-  pullSecret: 'Pull secret',
-  sshPublicKey: 'SSH public key',
-  SNODisclaimer: 'Single Node OpenShift disclaimer',
-  diskEncryptionTangServers: "Tang servers' URLs or thumbprints",
-  selectedHostIds: 'Hosts selection',
-  httpProxy: 'HTTP proxy',
-  httpsProxy: 'HTTPS proxy',
-  noProxy: 'No proxy',
-  machineNetworks: 'Machine networks',
-  clusterNetworks: 'Cluster networks',
-  serviceNetworks: 'Service networks',
-};
-
-export const HOST_VALIDATION_GROUP_LABELS: { [key in keyof ValidationsInfo]: string } = {
-  hardware: 'Hardware',
-  network: 'Network',
-  operators: 'Operators',
-  infrastructure: 'Infrastructure',
-};
-
-export const HOST_VALIDATION_LABELS: { [key in HostValidationId]: string } = {
-  'odf-requirements-satisfied': 'ODF requirements',
-  'disk-encryption-requirements-satisfied': 'Disk encryption requirements',
+export const clusterStatusLabels = (t: TFunction) => ({
```

Comment:
Let's keep the TS type definition of the data.
Same for the subsequent variables (eg. `clusterFieldLabels` etc)